pkg/puller/puller.go

```diff
@@ -149,22 +151,17 @@ func (p *Puller) manage(ctx context.Context, warmupDur time.Duration) {
 		}
 
 		p.recalcPeers(ctx, newRadius)
-
-		p.syncPeersMtx.Unlock()
 	}
 
-	tick := time.NewTicker(recalcPeersDur)
-	defer tick.Stop()
-
 	for {
+
+		onChange()
+
 		select {
 		case <-ctx.Done():
 			return
-		case <-tick.C:
-			onChange()
+		case <-time.After(recalcPeersDur):
```

Comment:
ticker is much more efficient than time.After. Although this is just 1 routine, it might do new allocation on every run of the loop. This can be easily avoided.

Any reason why this was changed?